include/ddc/kernels/splines/spline_evaluator_2d.hpp

```diff
@@ -70,16 +73,26 @@ class SplineEvaluator2DBatched
 private:
     spline_domain_type m_spline_domain;
 
+    LeftExtrapolationRule1 const& m_left1_bc;
+
+    RightExtrapolationRule1 const& m_right1_bc;
+
+    LeftExtrapolationRule2 const& m_left2_bc;
+
+    RightExtrapolationRule2 const& m_right2_bc;
```

Comment:
Would be ok if the class was not copied on the device
```suggestion
    LeftExtrapolationRule1 m_left1_bc;

    RightExtrapolationRule1 m_right1_bc;

    LeftExtrapolationRule2 m_left2_bc;

    RightExtrapolationRule2 m_right2_bc;
```